File: LC/3010_Divide_an_Array_Into_Subarrays_With_Minimum_Cost_I/solution.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
        int min_1 = INT_MAX, min_2 = INT_MAX;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] < min_1) {
                min_2 = min_1;
                min_1 = nums[i];
            } else if (nums[i] < min_2) {
                min_2 = nums[i];
            }
        }

        return nums[0] + min_1 + min_2;
    }
};

File: LC/3010_Divide_an_Array_Into_Subarrays_With_Minimum_Cost_I/solution.py
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        min_1, min_2 = inf, inf
        for i in range(1, len(nums)):
            if nums[i] < min_1:
                min_2 = min_1
                min_1 = nums[i]
            elif nums[i] < min_2:
                min_2 = nums[i]

        return nums[0] + min_1 + min_2